internal/controller/operator/instance/dapr_instance_controller_action_chart.go

```diff
@@ -0,0 +1,89 @@
+package instance
+
+import (
+	"context"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/dapr-sandbox/dapr-kubernetes-operator/api/operator/v1alpha1"
+	"github.com/dapr-sandbox/dapr-kubernetes-operator/pkg/controller/client"
+	"github.com/dapr-sandbox/dapr-kubernetes-operator/pkg/helm"
+	"github.com/go-logr/logr"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+)
+
+const (
+	ChartRepoUsernameKey = "username"
+	ChartRepoPasswordKey = "password"
+	ChartRepoEmbedded    = "embedded"
+)
+
+func NewChartAction(l logr.Logger) Action {
+	return &ChartAction{
+		engine: helm.NewEngine(),
+		l:      l.WithName("action").WithName("chart"),
+	}
+}
+
+type ChartAction struct {
+	engine *helm.Engine
+	l      logr.Logger
+}
+
+func (a *ChartAction) Configure(_ context.Context, _ *client.Client, b *builder.Builder) (*builder.Builder, error) {
+	return b, nil
+}
+
+func (a *ChartAction) Run(ctx context.Context, rc *ReconciliationRequest) error {
+	if rc.Resource.Status.Chart == nil {
+		rc.Resource.Status.Chart = &v1alpha1.ChartMeta{}
+	}
+
+	// TODO: maybe cache the chart
+	if rc.Resource.Spec.Chart != nil {
+		opts := helm.ChartOptions{}
+		opts.Name = rc.Resource.Spec.Chart.Name
+		opts.RepoURL = rc.Resource.Spec.Chart.Repo
+		opts.Version = rc.Resource.Spec.Chart.Version
+
+		if rc.Resource.Spec.Chart.Secret != "" {
```

Comment:
@ItalyPaleAle @salaboy 

I'm adding support to access private helm repos, is there any private dapr repo that I can use for testing purpose ?

Reply:
@lburgazzoli I am not aware of any, but maybe @ItalyPaleAle has Microsoft registries that we can use.